lib/combine_pdf/renderer.rb

```diff
@@ -123,15 +123,15 @@ def format_hash_to_pdf(object)
       # if the object is not a simple object, it is a dictionary
       # A dictionary shall be written as a sequence of key-value pairs enclosed in double angle brackets (<<...>>)
       # (using LESS-THAN SIGNs (3Ch) and GREATER-THAN SIGNs (3Eh)).
-      out << "<<\n".force_encoding(Encoding::ASCII_8BIT)
+      out << "<<\n".dup.force_encoding(Encoding::ASCII_8BIT)
       object.each do |key, value|
-        out << "#{object_to_pdf key} #{object_to_pdf value}\n".force_encoding(Encoding::ASCII_8BIT) unless PDF::PRIVATE_HASH_KEYS.include? key
+        out << "#{object_to_pdf key} #{object_to_pdf value}\n".dup.force_encoding(Encoding::ASCII_8BIT) unless PDF::PRIVATE_HASH_KEYS.include? key
       end
       object.delete :Length
-      out << '>>'.force_encoding(Encoding::ASCII_8BIT)
-      out << "\nstream\n#{object[:raw_stream_content]}\nendstream".force_encoding(Encoding::ASCII_8BIT) if object[:raw_stream_content]
+      out << '>>'.dup.force_encoding(Encoding::ASCII_8BIT)
+      out << "\nstream\n#{object[:raw_stream_content]}\nendstream".dup.force_encoding(Encoding::ASCII_8BIT) if object[:raw_stream_content]
       out << "\nendobj\n" if object[:indirect_reference_id]
-      out.join.force_encoding(Encoding::ASCII_8BIT)
+      out.join.dup.force_encoding(Encoding::ASCII_8BIT)
```

Comment:
Why `dup` the result?

Reply:
I searched for `.force_encoding` and replaced it with `.dup.force_encoding`. Perhaps a too aggressive/naive approach.

Reply:
Perhaps so... but perhaps not a bad idea.

I would consider automatically replacing all `'literal'.force_encoding(Encoding::ASCII_8BIT)` with `'literal'.b` ... an operation that could be automated by searching for `'.force_encoding(Encoding::ASCII_8BIT)` and replacing it with `'.b`